openml/extensions/sklearn/extension.py

```diff
@@ -934,12 +958,61 @@ def check(param_grid, restricted_parameter_name, legal_values):
                       '{GridSearchCV, RandomizedSearchCV}. '
                       'Should implement param check. ')
 
-            if not check(param_distributions, 'n_jobs', None):
+            if not SklearnExtension._check_parameter_value_recursive(param_distributions,
+                                                                     'n_jobs', None):
                 raise PyOpenMLError('openml-python should not be used to '
                                     'optimize the n_jobs parameter.')
 
+    def _can_measure_cputime(self, model: Any) -> bool:
+        """
+        Returns True if the parameter settings of model are chosen s.t. the model
+        will run on a single core (if so, openml-python can measure cpu-times)
+
+        Parameters:
+        -----------
+        model:
+            The model that will be fitted
+
+        Returns:
+        --------
+        bool:
+            True if all n_jobs parameters will be either set to None or 1, False otherwise
+        """
+        if not (
+                isinstance(model, sklearn.base.BaseEstimator) or self.is_hpo_class(model)
+        ):
+            raise ValueError('model should be BaseEstimator or BaseSearchCV')
+
         # check the parameters for n_jobs
-        return check(model.get_params(), 'n_jobs', [1, None])
+        return SklearnExtension._check_parameter_value_recursive(model.get_params(),
+                                                                 'n_jobs',
+                                                                 [1, None])
+
+    def _can_measure_wallclocktime(self, model: Any) -> bool:
+        """
+        Returns True if the parameter settings of model are chosen s.t. the model
+        will run on a preset number of cores (if so, openml-python can measure wall-clock time)
+
+        Parameters:
+        -----------
+        model:
+            The model that will be fitted
+
+        Returns:
+        --------
+        bool:
+            True if no n_jobs parameters is set to -1, False otherwise
+        """
+        if not (
+                isinstance(model, sklearn.base.BaseEstimator) or self.is_hpo_class(model)
+        ):
+            raise ValueError('model should be BaseEstimator or BaseSearchCV')
+
+        n_jobs_not_specified = \
+            SklearnExtension._check_parameter_value_recursive(model.get_params(), 'n_jobs', None)
+        n_jobs_is_minus_one = \
+            SklearnExtension._check_parameter_value_recursive(model.get_params(), 'n_jobs', [-1])
+        return n_jobs_not_specified or not n_jobs_is_minus_one
```

Comment:
There was a flake8 fail here (>100 characters), so I took the liberty of refactoring the code s.t. it complies and the variable names say enough (I think) so we don't need a 5-line block comment. I hope that's fine.